pkg/oc/cli/cmd/image/mirror/mirror.go

```diff
@@ -162,56 +173,109 @@ func parseDestination(ref string) (imageapi.DockerImageReference, DestinationTyp
 	var remainingArgs []string
-	overlap := make(map[string]string)
+	var mappings []Mapping
 	for _, s := range args {
 		parts := strings.SplitN(s, "=", 2)
 		if len(parts) != 2 {
 			remainingArgs = append(remainingArgs, s)
 			continue
 		}
 		if len(parts[0]) == 0 || len(parts[1]) == 0 {
-			return fmt.Errorf("all arguments must be valid SRC=DST mappings")
+			return nil, fmt.Errorf("all arguments must be valid SRC=DST mappings")
 		}
 		src, err := parseSource(parts[0])
 		if err != nil {
-			return err
+			return nil, err
 		}
 		dst, dstType, err := parseDestination(parts[1])
 		if err != nil {
-			return err
+			return nil, err
 		}
 		if _, ok := overlap[dst.String()]; ok {
-			return fmt.Errorf("each destination tag may only be specified once: %s", dst.String())
+			return nil, fmt.Errorf("each destination tag may only be specified once: %s", dst.String())
 		}
 		overlap[dst.String()] = src.String()
 
-		o.Mappings = append(o.Mappings, Mapping{Source: src, Destination: dst, Type: dstType})
+		mappings = append(mappings, Mapping{Source: src, Destination: dst, Type: dstType})
 	}
 
 	switch {
-	case len(remainingArgs) == 0 && len(o.Mappings) > 0:
-		// user has input arguments
-	case len(remainingArgs) > 1 && len(o.Mappings) == 0:
+	case len(remainingArgs) > 1 && len(mappings) == 0:
 		src, err := parseSource(remainingArgs[0])
 		if err != nil {
-			return err
+			return nil, err
 		}
 		for i := 1; i < len(remainingArgs); i++ {
 			dst, dstType, err := parseDestination(remainingArgs[i])
 			if err != nil {
-				return err
+				return nil, err
 			}
 			if _, ok := overlap[dst.String()]; ok {
-				return fmt.Errorf("each destination tag may only be specified once: %s", dst.String())
+				return nil, fmt.Errorf("each destination tag may only be specified once: %s", dst.String())
 			}
 			overlap[dst.String()] = src.String()
-			o.Mappings = append(o.Mappings, Mapping{Source: src, Destination: dst, Type: dstType})
+			mappings = append(mappings, Mapping{Source: src, Destination: dst, Type: dstType})
 		}
-	case len(remainingArgs) == 1 && len(o.Mappings) == 0:
-		return fmt.Errorf("all arguments must be valid SRC=DST mappings, or you must specify one SRC argument and one or more DST arguments")
-	default:
+	case len(remainingArgs) == 1 && len(mappings) == 0:
+		return nil, fmt.Errorf("all arguments must be valid SRC=DST mappings, or you must specify one SRC argument and one or more DST arguments")
+	}
+	return mappings, nil
+}
+
+func parseFile(filename string, overlap map[string]string) ([]Mapping, error) {
+	var fileMappings []Mapping
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+	lineNumber := 1
+	for s.Scan() {
+		line := s.Text()
+
+		// remove comments and whitespace
+		if i := strings.Index(line, "#"); i != -1 {
+			line = line[0:i]
+		}
+		line = strings.TrimSpace(line)
+		if len(line) == 0 {
+			continue
```

Comment:
It will go to the next line without incrementing lineNumber.

Reply:
thanks